xla/literal_comparison_test.cc

```diff
@@ -25,173 +25,115 @@ limitations under the License.
 namespace xla {
 namespace {
 
-TEST(LiteralComparisonTest, F8E4M3FNCompareNear_Equal) {
-  auto actual =
-      LiteralUtil::CreateR0<tsl::float8_e4m3fn>(tsl::float8_e4m3fn(8.0));
-  auto expected =
-      LiteralUtil::CreateR0<tsl::float8_e4m3fn>(tsl::float8_e4m3fn(8.0));
+template <typename T>
+struct TypeTraits {
+  static constexpr float expected_1ulp = 9.0;
+  static constexpr int within_n_values_1ulp = 1;
+  static constexpr float expected_4ulp = 12.0;
+  static constexpr float expected_using_float_4ulp = 12.1;
```

Comment:
It's unclear what these values are without reading the tests. Instead, I think we should probably define this values inline with the tests, instead of having type traits. For example, we could have CompareNear_NotEqual_1ulp start with

```c++
TYPED_TEST(LiteralComparisonTest, CompareNear_NotEqual_1ulp) {
  PrimitiveType type = GetPrimitiveType();
  TypeParam 1ulp_from_8 = type == F8E5M2 ? TypeParam{10} : TypeParam{9};
  auto actual = LiteralUtil::CreateR0<TypeParam>(TypeParam(8.0));
  auto expected = LiteralUtil::CreateR0<TypeParam>(1ulp_from_8);
  ...
```

Reply:
removed TypeTraits. Inlined value selection looks much better.